big_transfer.py

```diff
@@ -20,61 +20,98 @@ class BigTransferEncoder(Executor):
+    If it does exist, the model in this folder is used.
+    If it does not exist, the model specified in the model_name will be
+    downloaded into this path and the downloaded model is used.
 
-        .. highlight:: bash
-         .. code-block:: bash
+    In the end, the `model_path` should be a directory path,
+    which has the following structure:
 
-            .
-            ├── saved_model.pb
-            └── variables
-                ├── variables.data-00000-of-00001
-                └── variables.index
+    .. highlight:: bash
+     .. code-block:: bash
+        .
+        ├── saved_model.pb
+        └── variables
+            ├── variables.data-00000-of-00001
+            └── variables.index
 
-        :param channel_axis: the axis id of the channel, -1 indicate the color
-            channel info at the last axis. If given other, then `
-            `np.moveaxis(data, channel_axis, -1)`` is performed before :meth:`encode`.
+    :param channel_axis: the axis id of the channel, -1 indicate the color
+        channel info at the last axis. If given other, then `
+        `np.moveaxis(data, channel_axis, -1)`` is performed before :meth:`encode`.
+
+    :param: on_gpu: If true, the GPU will be used. Make sure to have
+        tensorflow-gpu==2.5 installed
     """
     def __init__(self,
                  model_path: Optional[str] = 'pretrained',
+                 model_name: Optional[str] = 'R50x1',
                  channel_axis: int = 1,
                  on_gpu: bool = False,
                  *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.channel_axis = channel_axis
         self.model_path = model_path
+        self.model_name = model_name
         self.on_gpu = on_gpu
         self.logger = default_logger
 
-        if self.model_path and os.path.exists(self.model_path):
-            cpus = tf.config.experimental.list_physical_devices(
-                device_type='CPU')
-            gpus = tf.config.experimental.list_physical_devices(
-                device_type='GPU')
-            if self.on_gpu and len(gpus) > 0:
-                cpus.append(gpus[0])
-            tf.config.experimental.set_visible_devices(devices=cpus)
-            self.logger.info(f'BiT model path: {self.model_path}')
-            from tensorflow.python.keras.models import load_model
-            _model = load_model(self.model_path)
-            self.model = _model.signatures['serving_default']
-            self._get_input = tf.convert_to_tensor
-        else:
-            raise PretrainedModelFileDoesNotExist(
-                f'model at {self.model_path} does not exist')
+        if not os.path.exists(self.model_path):
+            self.download_model()
+
+        cpus = tf.config.experimental.list_physical_devices(
+            device_type='CPU')
+        gpus = tf.config.experimental.list_physical_devices(
+            device_type='GPU')
+        if self.on_gpu and len(gpus) > 0:
+            cpus.append(gpus[0])
```

Comment:
Not sure, but should we raise an Exception, if `self.on_gpu and len(gpus) == 0`?

Reply:
I did a warning here: https://github.com/jina-ai/executor-big-transfer-encoder/pull/1/commits/0f11e821d6f2ef4b8b690a273a8c0aae0ebfaedc